R/pkg/R/DataFrame.R

```diff
@@ -392,7 +392,11 @@ setMethod("coltypes",
                 }
 
                 if (is.null(type)) {
-                  stop(paste("Unsupported data type: ", x))
+                  specialtype <- specialtypeshandle(x)
```

Comment:
and wait, we are passing `x` to `specialtypeshandle`?
L395 `specialtype <- specialtypeshandle(x)`
isn't `x` the `SparkDataFrame`??